File: src/test/java/br/com/zumbolovsky/fateapp/PostgreSQLSetup.java
package br.com.zumbolovsky.fateapp;

import org.testcontainers.containers.PostgreSQLContainer;

public class PostgreSQLSetup {
    private static final String image = "postgres:latest";
    private static final Integer port = 5432;

    private static PostgreSQLContainer<?> container() {
        final PostgreSQLContainer<?> postgreSQLContainer = new PostgreSQLContainer<>(image);
        postgreSQLContainer.withExposedPorts(port);
        return postgreSQLContainer;
    }

    public PostgreSQLSetup() {
        final PostgreSQLContainer<?> container = container();
        container.start();
        System.setProperty("spring.datasource.url", container.getJdbcUrl());
        System.setProperty("spring.datasource.username", container.getUsername());
        System.setProperty("spring.datasource.password", container.getPassword());
    }
}

File: src/test/kotlin/br/com/zumbolovsky/fateapp/PostgreSQLSetup.kt
package br.com.zumbolovsky.fateapp

import org.testcontainers.containers.PostgreSQLContainer

class PostgreSQLSetup {
    private val image = "postgres:latest"
    private val port = 5432

    private val container =
        PostgreSQLContainer(image)
            .withExposedPorts(port)

    private val jdbcUrl: String
    get() = container.jdbcUrl

    private val username: String
    get() = container.username

    private val password: String
    get() = container.password

    init {
        container.start()
        System.setProperty("spring.datasource.url", jdbcUrl)
        System.setProperty("spring.datasource.username", username)
        System.setProperty("spring.datasource.password", password)
    }
}
